### PR TITLE
Don't trigger `no-incorrect-deep-equal` for regular expression

### LINKED ISSUE #270: Regular expression in `t.deepEqual()` shouldn't trigger warning
[Issuehunt badges](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/270)

Rule: [no-incorrect-deep-equal](https://github.com/avajs/eslint-plugin-ava/blob/master/docs/rules/no-incorrect-deep-equal.md)

Example:
```js
const ext2regexp = require('ext-to-regexp');

const jsx = ext2regexp('js', 'jsx');

t.deepEqual(jsx, /\.(js|jsx)$/); //  shouldn't trigger warning
```

---

> *Note: This issue has a bounty, so it's expected that you are an experienced programmer and that you give it your best effort if you intend to tackle this. Don't forget, if applicable, to add tests, docs (double-check for typos). And don't be sloppy. Review your own diff multiple times and try to find ways to improve and simplify your code. Instead of asking too many questions, present solutions. The point of an issue bounty is to reduce my workload, not give me more. Include a 🦄 in your PR description to indicate that you've read this. Thanks for helping out 🙌* - @sindresorhus

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)
### Submitted pull Requests
- [#276 do not trigger no-incorrect-deep-equal for regular expression](https://issuehunt.io/r/avajs/eslint-plugin-ava/pull/276)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/avajs/eslint-plugin-ava/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by mikaoelitiana**
fixes #270

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#270: Regular expression in `t.deepEqual()` shouldn't trigger warning](https://issuehunt.io/repos/49722492/issues/270)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: test/no-incorrect-deep-equal.js [hunk 1]
```diff
@@ -40,6 +40,12 @@ ruleTester.run('no-incorrect-deep-equal', rule, {
 			test('x', t => {
 				t.notDeepEqual(expression, []);
 			});
+		`,
+		`
+			${header}
+			test('x', t => {
+				t.deepEqual(expression, /regex/);
```
Comment (sindresorhus) [inline test/no-incorrect-deep-equal.js]: Also add a test with the arguments inversed: `t.deepEqual(/regex/, expression);`

Comment (mikaoelitiana) [reply test/no-incorrect-deep-equal.js]: Ok I will do that

**Comment by sindresorhus** [comment]
Travis is failing.

**Comment by mikaoelitiana** [comment]
It look like it is a general issue on integration test, I will check if I can fix this

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**